examples/qec.qasm

```diff
@@ -7,11 +7,11 @@ qubit a[2];
 bit c[3];
 bit syn[2];
 
-def syndrome qubit[3]:d, qubit[2]:a -> bit[2] { 
+def syndrome qubit[3]:d, qubit[2]:a -> bit[2] {
```

Comment:
parantheses missing

Reply:
Do qubits go inside or outside the parens on subroutines?

Reply:
This is a good point and I am incorrect according to the current specification. I know we had previously had a discussion on unifying classical and quantum arguments within parentheses in subroutines and I thought the outcome had been in favour but the document doesn't reflect this. 

I am no longer sure how I feel on this subject and I will resolve these comments.

Reply:
I've spent some time thinking about this today after reading some feedback from @ndebeaudrap and I am no longer opposed to leaving qubits out of brackets in the spirit of backward compatibility with OpenQASM2. It does make differentiating between gates and subroutines more difficult.

Reply:
I hadn't replied yet because of the potential ambiguity and wanted to see how antlr might parse it (which I haven't gotten to do yet). I feel it is reasonable to leave qubits out of parentheses but require classical arguments to be in the parentheses. If there are no classical arguments we can ignore the parenthesis. It is true in this case that you can't distinguish between kernel, subroutine or gate calls. However, if you put everything in parentheses for subroutines, you would still have ambiguity on whether it is a subroutine or a kernel. 

I don't think you necessarily need to be clear which is which at call time. This is similar to not knowing whether a gate has a `defcal` or a unitary `gate` implementation.

Reply:
I agree with the first paragraph but 

> I don't think you necessarily need to be clear which is which at call time. This is similar to not knowing whether a gate has a defcal or a unitary gate implementation.

Defcals are never called, they are a root level gate implementation passed to the compiler. Calls to defcals are indirectly performed through gates, and linked by the compiler. Only gates/measure/reset (operations) and subroutines/kernels are called.

Reply:
Resolving, the current spec is correct.